File: backend/app/schemas/inspection.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class InspectionCreate(BaseModel):
    property_name: str = Field(
        ...,
        min_length=1,
        max_length=200,
    )

    inspection_type: Literal[
        "move_in",
        "move_out",
    ]


class PredictionSummary(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    label: str
    confidence: float
    severity: str
    created_at: datetime


class ImageSummary(BaseModel):
    id: int
    image_url: str
    uploaded_at: datetime
    prediction: PredictionSummary | None = None


class RoomSummary(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    name: str
    images: list[ImageSummary]


class InspectionResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    user_id: int
    property_name: str
    inspection_type: str
    status: str


class InspectionDetailResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    user_id: int
    property_name: str
    inspection_type: str
    status: str
    created_at: datetime
    rooms: list[RoomSummary]

File: frontend/src/types/inspection.ts
export type InspectionType =
  | "move_in"
  | "move_out";


export type Severity =
  | "None"
  | "Low"
  | "Medium"
  | "High";


export interface Prediction {
  id: number;
  label: string;
  confidence: number;
  severity: Severity;
  created_at: string;
}


export interface InspectionImage {
  id: number;
  image_url: string;
  uploaded_at: string;
  prediction: Prediction | null;
}


export interface Room {
  id: number;
  name: string;
  images: InspectionImage[];
}


export interface Inspection {
  id: number;
  property_name: string;
  inspection_type: InspectionType;
  status: string;
  created_at: string;
  rooms: Room[];
}


export interface CreatedInspection {
  id: number;
  property_name: string;
  inspection_type: InspectionType;
  status: string;
}


export interface UploadedImageResponse {
  image: {
    id: number;
    room_id: number;
    file_path: string;
    uploaded_at: string;
  };

  prediction: Prediction;

  top_predictions: {
    class: string;
    confidence: number;
  }[];
}